### DISCUSSION TITLE
Synchronize Stored_Elastic_Energy_Density field

### DISCUSSION THREAD

**Discussion by jritta** [peridigm/peridigm#247] [Q&A]
Dear community,

I want to synchronize the field `Stored_Elastic_Energy_Density` throughout multiple MPI processes. The common fields like `Force_Density` or `Coordinates` are synchronized by default. Furthermore for the damage and material models the possibility is availability to synchronize by `setFieldIdsToSyncronizeAfterInitialize` and `setFieldIdsToSyncronizeAfterPrecompute`.

What is the best way to synchronize the field `Stored_Elastic_Energy`? It is created by the compute class and thus not available when the constructor of the material model is called. Thus it is not possible to use the way via `setFieldIdsToSyncronizeAfterPrecompute`.

Best regards,
Jonas

**Comment by oldninja** [comment]
I had the same question a few months ago. Dear @johntfoster / @djlittl, what would be the best way to approach this?

**Reply by djlittl** [reply]
Hmm, tricky.

Regarding the comment from @jritta that Stored_Elastic_Energy isn’t available when the material model constructor is called, you might try creating the field in the material model constructor itself.  I’m not 100% sure this will work, but it’s worth a try.  Just make sure that you define the field the exact same way that it is defined by the compute class.  Peridigm will throw an error if you have two different fields with the same name.  But it is okay with having any given field defined more than once, as long as all the definitions match exactly.

I you could create it in the material model constructor, then you could pass it to one of the setFieldIdsToSyncronize options.

**Reply by jritta** [reply]
Thank you very much for your quick reply! I implemented into the ComputeManager a similar function called `setFieldIdsToSynchronizeAfterCompute`. This way of doing seems to do the job. Unfortunately, these days I have not the time for a proper and clean implementation that is worth to make a pull request.

@djlittl Thank you so much for your support!

Best,
Jonas
